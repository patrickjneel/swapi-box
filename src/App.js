import React, { Component } from 'react';
import './App.css';
import Scrolling from './components/scrolling-text/Scrolling-text';
import Header from './components/header/Header';
import CardContainer from './components/cardContainer/CardContainer';
import API from './components/api/API';

const {fetchFilm, fetchPeople, fetchVehicles, fetchPlanets, planetResidents} = API;

class App extends Component {
  constructor() {
    super();
    this.state = {
      film: {},
      people: [],
      vehicles: [],
      planets: [],
      location: 'home',
      favorites: []
  
    }
      this.upDateData = this.upDateData.bind(this);
      this.updateFavorite = this.updateFavorite.bind(this);
      this.showHome = this.showHome.bind(this);
  }

  upDateData(newData) {
    this.setState({location: newData})
  }

  showHome() {
    console.log('click')
    return (
      <div className="container">
        <img src={require("./at-at.gif")} />
      </div>
    )
  }

  updateFavorite(card) {

    let favorites = this.state.favorites;
    
    const findInFavorites = favorites.find(favorite => favorite.name === card.name)
    if(findInFavorites) {
      favorites = favorites.filter(favorite => favorite.name !== card.name)
    } else {
      favorites.push(card)
    }
    this.setState({favorites})
  }

  async componentDidMount() {
    const film = await fetchFilm();
    const people =  await fetchPeople()
    const vehicles = await fetchVehicles();
    const planets =  await fetchPlanets();
    this.setState({ film, people, vehicles, planets })
  }

  render() {
    const arrayToRender = this.state.location;

    return (
      <div className="App">
      <div className="top">
        <Scrolling 
          film={this.state.film}
        />
      </div>
      <div className="bottom">
        <Header 
          upDateData={this.upDateData}
          favorites={this.state.favorites.length}
          location={this.state.location}
          showHome={this.showHome}
        />
        <CardContainer 
          itemData={this.state[arrayToRender]}
          location={this.state.location}
          updateFavorite={this.updateFavorite}
          favorites={this.state.favorites} 
        />
      </div>
      </div>
    );
  }
}

export default App;
