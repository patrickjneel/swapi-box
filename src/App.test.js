import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';


jest.mock('./__mocks__/fetchAPIDATA');

describe('App Test', () => {

  it('should be defined', () => {
  const renderedApp = shallow(<App />, {disableLifeCycleMethods: true});

  expect(renderedApp).toBeDefined();
    
  });

  it('should render the correct components', () => {
    const renderedApp = shallow(<App />)
    const foundHeader = renderedApp.find('Header');
    const foundConatiner = renderedApp.find('CardContainer')
    const foundScroll = renderedApp.find('Scrolling')

    expect(foundHeader.length).toEqual(1)
    expect(foundConatiner.length).toEqual(1)
    expect(foundScroll.length).toEqual(1)
  })
  
  it('should have a default state', () => {
    const renderedApp = shallow(<App />)
    const expectedFilm = {};
    const expectedPeople = [];
    const expectedVehicles = [];
    const expectedPlanets = [];
    const expectedLocation = 'people';
    const expectedFavorites = [] 

    expect(renderedApp.state('film')).toEqual(expectedFilm)
    expect(renderedApp.state('people')).toEqual(expectedPeople)
    expect(renderedApp.state('vehicles')).toEqual(expectedVehicles)
    expect(renderedApp.state('planets')).toEqual(expectedPlanets)
    expect(renderedApp.state('location')).toEqual(expectedLocation)
    expect(renderedApp.state('favorites')).toEqual(expectedFavorites)

  })

  it('should match the snap shot', () => {
    const renderedApp = shallow(<App />);

    expect(renderedApp).toMatchSnapshot();
  })

  it('should set the location when the vehicle is passed in', () => {
    const renderedApp = shallow(<App />);
    const expectedLocation = 'vehicles';

    renderedApp.instance().upDateData('vehicles');
    expect(renderedApp.state('location')).toEqual(expectedLocation);
  })

  it('should set the people location when  people is passed in', () => {
    const renderedApp = shallow(<App />);
    const expectedLocation = 'people';

    renderedApp.instance().upDateData('people');
    expect(renderedApp.state('location')).toEqual(expectedLocation);
  })

  it('should set the planets location when the planets is passed in', () => {
    const renderedApp = shallow(<App />);
    const expectedLocation = 'planets';

    renderedApp.instance().upDateData('planets');
    expect(renderedApp.state('location')).toEqual(expectedLocation);
  })

  it('should set the planets when the favorites is called', () => {
    const renderedApp = shallow(<App />);
    const expectedLocation = 'favorites';

    renderedApp.instance().upDateData('favorites');
    expect(renderedApp.state('location')).toEqual(expectedLocation);
  })

  it('should add a favorite when a card is clicked', () => {
    const renderedApp = shallow(<App />);
    const initialState = [];
    const favoritedCard = 
       {
         homeworld: "Stewjon",   
         name: "Obi-Wan Kenobi",
         population: "unknown",
         species: "Human"
       }
     const expectedState = [{
         homeworld: "Stewjon",   
         name: "Obi-Wan Kenobi",
         population: "unknown",
         species: "Human"
       }]

    expect(renderedApp.state('favorites')).toEqual(initialState);
    renderedApp.instance().updateFavorite(favoritedCard);
    expect(renderedApp.state('favorites')).toEqual(expectedState);

  })

  it('should add and remove a card from favorites when clicked twice', () => {
    const renderedApp = shallow(<App />);
    const initialState = [];
    // const 
  })





  // it.skip('should add a favorite card to favorties when card has been clicked' , () => {
  //   const renderedApp = shallow(<App />);
  //   const initialState = [];
  //   const mockData = [
  //     {
  //       name: "Luke Skywalker",
  //       homeworld: "Tatooine",
  //       population: "200000",
  //       species: "Human"
  //     }
  //   ]
  //   expect(renderedApp.state('favorites')).toEqual(initialState);
  //     console.log(renderedApp.debug())
  //     renderedApp.setState({ favorites: mockData })

  //     renderedApp.instance().updateFavorite()

  //   expect(renderedApp.state('favorites')).toEqual(mockData)



  // });
});
