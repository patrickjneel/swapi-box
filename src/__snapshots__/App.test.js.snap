// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should match the snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="top"
>
          <Scrolling
                    film={Object {}}
          />
</div>,
        <div
          className="bottom"
>
          <Header
                    favorites={0}
                    location="people"
                    upDateData={[Function]}
          />
          <CardContainer
                    favorites={Array []}
                    itemData={Array []}
                    location="people"
                    updateFavorite={[Function]}
          />
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Scrolling
            film={Object {}}
/>,
          "className": "top",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "film": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              favorites={0}
              location="people"
              upDateData={[Function]}
/>,
            <CardContainer
              favorites={Array []}
              itemData={Array []}
              location="people"
              updateFavorite={[Function]}
/>,
          ],
          "className": "bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "favorites": 0,
              "location": "people",
              "upDateData": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "favorites": Array [],
              "itemData": Array [],
              "location": "people",
              "updateFavorite": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="top"
>
            <Scrolling
                        film={Object {}}
            />
</div>,
          <div
            className="bottom"
>
            <Header
                        favorites={0}
                        location="people"
                        upDateData={[Function]}
            />
            <CardContainer
                        favorites={Array []}
                        itemData={Array []}
                        location="people"
                        updateFavorite={[Function]}
            />
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Scrolling
              film={Object {}}
/>,
            "className": "top",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "film": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                favorites={0}
                location="people"
                upDateData={[Function]}
/>,
              <CardContainer
                favorites={Array []}
                itemData={Array []}
                location="people"
                updateFavorite={[Function]}
/>,
            ],
            "className": "bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "favorites": 0,
                "location": "people",
                "upDateData": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "favorites": Array [],
                "itemData": Array [],
                "location": "people",
                "updateFavorite": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
